src/main/kotlin/com/sourcegraph/cody/edit/FixupService.kt

```diff
@@ -0,0 +1,166 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.config.ConfigUtil.isCodyEnabled
+import com.sourcegraph.utils.CodyEditorUtil
+
+/** Controller for commands that allow the LLM to edit the code directly. */
+@Service(Service.Level.PROJECT)
+class FixupService(val project: Project) : Disposable {
+  private val logger = Logger.getInstance(FixupService::class.java)
+
+  // We only use this for multiplexing task updates from the Agent to concurrent sessions.
+  // TODO: Consider doing the multiplexing in CodyAgentClient instead.
+  private var activeSessions: MutableMap<String, FixupSession> = mutableMapOf()
```

Comment:
Is it possible to have more than one fixup running at the same time, e.g. in different chat window?
If so, this most likely need to be changed to be multithread-safe.

Reply:
It is, and there's a tracking bug to fix all the concurrency issues with the existing code. (Or, if we don't have time to get it done for release, we need to block concurrent edit tasks.)